study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/StudyBuilder.java

```diff
@@ -624,6 +627,12 @@ default int invalidateMailingAddressStatuses(long studyId) {
                 + "  where address_id in (<addressIds>)")
         int updateAddressStatusesToInvalid(@BindList(value = "addressIds", onEmpty = BindList.EmptyHandling.NULL) Set<Long> addressIds);
 
+        @SqlUpdate("update kit_schedule_record as r"
+                + "   join kit_configuration as k on k.kit_configuration_id = r.kit_configuration_id"
+                + "    set r.opted_out = true, r.num_occurrences = 100"
+                + "  where k.study_id = :studyId")
+        int invalidateKitScheduleRecords(@Bind("studyId") long studyId);
```

Comment:
To invalidate: We mark as user opted out and is the 100 here to pretend as 100 occurrences happened ? If not whats the significance of updating to 100 ?

Reply:
It's not significant.

Reply:
I put in a ridiculously big number here to avoid people getting potentially confused on whether it was truly opted-out or not.

Reply:
ok got it. Its part of study invalidation